### PR TITLE
IOS-XR: Add [iosxr_lldp_global] module

### PR DESCRIPTION
**Pull request by NilashishC**
Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>
Depends-On: https://github.com/ansible/ansible/pull/59153

##### SUMMARY
- Adds iosxr_lldp_global resource module.
- Targets - https://github.com/ansible-network/resource_module_models/pull/47

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
iosxr_lldp_global.py

### CODE REVIEW

File: lib/ansible/module_utils/network/iosxr/config/lldp_global/lldp_global.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    ]
+
+    gather_network_resources = [
+        'lldp_global',
+    ]
+
+    def __init__(self, module):
+        super(Lldp_global, self).__init__(module)
+
+    def get_lldp_facts(self):
```
Comment (ganeshrn) [inline lib/ansible/module_utils/network/iosxr/config/lldp_global/lldp_global.py]: As per naming convention, the name is `get_lldp_global_facts`

File: lib/ansible/module_utils/network/iosxr/config/lldp_global/lldp_global.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        warnings = list()
+        commands = list()
+
+        existing_lldp_facts = self.get_lldp_facts()
+        commands.extend(self.set_config(existing_lldp_facts))
```
Comment (ganeshrn) [inline lib/ansible/module_utils/network/iosxr/config/lldp_global/lldp_global.py]: existing_lldp_global_facts

File: lib/ansible/module_utils/network/iosxr/config/lldp_global/lldp_global.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            if not self._module.check_mode:
+                self._connection.edit_config(commands)
+            result['changed'] = True
+        result['commands'] = commands
+
+        changed_lldp_facts = self.get_lldp_facts()
+
+        result['before'] = existing_lldp_facts
+        if result['changed']:
+            result['after'] = changed_lldp_facts
```
Comment (ganeshrn) [inline lib/ansible/module_utils/network/iosxr/config/lldp_global/lldp_global.py]: changed_lldp_global_facts

File: lib/ansible/modules/network/iosxr/iosxr_lldp_global.py [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+            type: bool
+  state:
+    description:
+      - The state the configuration should be left in.
+    type: str
+    choices:
+    - merged
+    - replaced
+    - deleted
+    default: merged
```
Comment (ganeshrn) [inline lib/ansible/modules/network/iosxr/iosxr_lldp_global.py:103]: Add the note mentioning the version of iosxr test against and connection type tp use with his module and platform guide reference.
 Refer: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/junos/junos_interfaces.py#L106-L107

Comment (ganeshrn) [reply lib/ansible/modules/network/iosxr/iosxr_lldp_global.py:103]: Applicable for other iosxr resource modules as well

File: lib/ansible/modules/network/iosxr/iosxr_lldp_global.py [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+short_description: Manage Global Link Layer Discovery Protocol (LLDP) settings on IOS-XR devices.
+description:
+  - This module manages Global Link Layer Discovery Protocol (LLDP) settings on IOS-XR devices.
+author: Nilashish Chakraborty (@NilashishC)
+options:
+  config:
+    description: The provided global LLDP configuration.
+    type: dict
+    suboptions:
+      holdtime:
```
Comment (justjais) [inline lib/ansible/modules/network/iosxr/iosxr_lldp_global.py:54]: for this module, none of the params are mentioned as `required`, is that expected?

File: lib/ansible/modules/network/iosxr/iosxr_lldp_global.py (hunk 4 above)
Comment (NilashishC) [reply lib/ansible/modules/network/iosxr/iosxr_lldp_global.py:103]: @ganeshrn it seems we don't have platform guide reference for IOS-XR

File: lib/ansible/modules/network/iosxr/iosxr_lldp_global.py (hunk 5 above)
Comment (NilashishC) [reply lib/ansible/modules/network/iosxr/iosxr_lldp_global.py:54]: We should probably fail, if state is not `deleted` and `config` is empty.

**Comment by NilashishC** [comment]
One more `Socket is closed` issue.

**Comment by NilashishC** [comment]
recheck

**Review by danielmellado — COMMENTED** [review]
Hi @NilashishC , overall LGTM, my only concern is the dependency on the WIP change. I'd merge that one before doing so with this. Thanks!

**Review by ganeshrn — APPROVED** [review]
lgtm

**Review by danielmellado — APPROVED** [review]
LGTM, with the caveat of the WIP which it depeds on, let's not forget about it ;)

**Comment by NilashishC** [comment]
> LGTM, with the caveat of the WIP which it depeds on, let's not forget about it ;)

@danielmellado true. We _should_ keep a close eye on the nightly runs. Thanks for the +1!

**Pull request merged**